scripts/bigquery.js

```diff
@@ -81,6 +108,181 @@ async function createWebsitesTable() {
 	)
 }
 
+async function createAdUnitsTable() {
+	// Create the dataset
+	await dataset.createTable(ADUNITS_TABLE_NAME, {
+		schema: {
+			fields: [
+				{ name: 'id', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'type', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'title', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'created', type: 'TIMESTAMP', mode: 'REQUIRED' },
+				{ name: 'owner', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'mediaUrl', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'mediaMime', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'targetUrl', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'archived', type: 'BOOLEAN', mode: 'NULLABLE' },
+				{ name: 'modified', type: 'TIMESTAMP', mode: 'NULLABLE' },
+			],
+		},
+	})
+	return startImport(
+		ADUNITS_TABLE_NAME,
+		getMongo()
+			.collection('adUnits')
+			.find()
+			.sort({ _id: -1 })
+			.stream(),
+		function(adUnit) {
+			if (!adUnit) return
+			return {
+				id: adUnit.ipfs,
+				type: adUnit.type,
+				title: adUnit.title,
+				owner: adUnit.owner,
+				created: dateToTimestamp(adUnit.created),
+				mediaUrl: ipfsSrc(adUnit.mediaUrl),
+				mediaMime: adUnit.mediaMime,
+				targetUrl: adUnit.targetUrl,
+				archived: adUnit.archived,
+				modified: dateToTimestamp(adUnit.modified),
+			}
+		}
+	)
+}
+
+async function createAudiencesTable() {
+	// Create the dataset
+	await dataset.createTable(AUDIENCES_TABLE_NAME, {
+		schema: {
+			fields: [
+				{ name: 'id', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'campaignId', type: 'STRING', mode: 'NULLABLE' },
+				{ name: 'title', type: 'STRING', mode: 'NULLABLE' },
+				{ name: 'created', type: 'TIMESTAMP', mode: 'REQUIRED' },
+				{ name: 'owner', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'locationApply', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'locationShowIn', type: 'STRING', mode: 'REPEATED' },
+				{ name: 'locationNotShowIn', type: 'STRING', mode: 'REPEATED' },
+				{ name: 'categoriesApply', type: 'STRING', mode: 'REPEATED' },
+				{ name: 'categoriesShowIn', type: 'STRING', mode: 'REPEATED' },
+				{ name: 'categoriesNotShowIn', type: 'STRING', mode: 'REPEATED' },
+				{ name: 'publishersApply', type: 'STRING', mode: 'REQUIRED' },
+				{ name: 'publishersShowIn', type: 'STRING', mode: 'REPEATED' },
+				{ name: 'publishersNotShowIn', type: 'STRING', mode: 'REPEATED' },
+				{ name: 'disableFrequencyCapping', type: 'BOOLEAN', mode: 'NULLABLE' },
+				{ name: 'includeIncentivized', type: 'BOOLEAN', mode: 'NULLABLE' },
+				{
+					name: 'limitDailyAverageSpending',
+					type: 'BOOLEAN',
+					mode: 'NULLABLE',
+				},
+			],
+		},
+	})
+	return startImport(
+		AUDIENCES_TABLE_NAME,
+		getMongo()
+			.collection('audiences')
+			.find()
```

Comment:
Cant you add filter `{campaignId: { $exists: true, $ne: null }}` instead `if (!campaign) return // skip if not used in campaign`